eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelData.scala

```diff
@@ -62,9 +57,37 @@ case object CLOSED extends ChannelState
 case object OFFLINE extends ChannelState
 case object SYNCING extends ChannelState
 case object WAIT_FOR_REMOTE_PUBLISH_FUTURE_COMMITMENT extends ChannelState
-case object ERR_FUNDING_LOST extends ChannelState
 case object ERR_INFORMATION_LEAK extends ChannelState
 
+sealed trait ChannelStateData {
+  def channelId: ByteVector32
+  def channelData(): Option[PersistentChannelData] = this match {
+    case _: DATA_WAIT_FOR_INIT_INTERNAL => None
+    case DATA_OPENING(_, _, Some(data: PersistentChannelData)) => Some(data)
+    case DATA_OPENING(_, _, _) => None
+    case DATA_NORMAL(data) => Some(data)
+    case DATA_SHUTDOWN(data) => Some(data)
+    case DATA_NEGOTIATING(data) => Some(data)
+    case DATA_CLOSING(data) => Some(data)
+    case DATA_CLOSED(data_opt) => data_opt
+    case DATA_OFFLINE(data) => Some(data)
+    case DATA_SYNCING(data) => Some(data)
+    case DATA_WAIT_FOR_REMOTE_PUBLISH_FUTURE_COMMITMENT(data) => Some(data)
+    case DATA_ERR_INFORMATION_LEAK(data) => Some(data)
+  }
+}
+case class DATA_WAIT_FOR_INIT_INTERNAL() extends ChannelStateData { val channelId: ByteVector32 = ByteVector32.Zeroes }
+case class DATA_OPENING(actor: ActorRef, channelId: ByteVector32, data_opt: Option[ChannelOpenData]) extends ChannelStateData
+case class DATA_NORMAL(data: ChannelData.Normal) extends ChannelStateData { val channelId: ByteVector32 = data.channelId }
+case class DATA_SHUTDOWN(data: ChannelData.ShuttingDown) extends ChannelStateData { val channelId: ByteVector32 = data.channelId }
+case class DATA_NEGOTIATING(data: ChannelData.Negotiating) extends ChannelStateData { val channelId: ByteVector32 = data.channelId }
+case class DATA_CLOSING(data: ChannelData.Closing) extends ChannelStateData { val channelId: ByteVector32 = data.channelId }
+case class DATA_CLOSED(data_opt: Option[PersistentChannelData]) extends ChannelStateData { val channelId: ByteVector32 = data_opt.map(_.channelId).getOrElse(ByteVector32.Zeroes) }
```

Comment:
We can't, because since it's not persistent we wouldn't be able to load it back from the DB, whereas we have the ability to load closed channels from the DB after a restart

Reply:
> whereas we have the ability to load closed channels from the DB after a restart

But we didn't before this PR right?